src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -314,7 +340,7 @@ public InetSocketAddress next(long spinDelay) {
                 addr = nextHostInReconfigMode();
                 if (addr != null) {
                 	currentIndex = serverAddresses.indexOf(addr);
-                	return addr;                
+                	return resolve(addr);
```

Comment:
Can you do some test to find out performance/latency of resolving an addr? If it's costly, maybe we should cache the resolved one instead of resolve it every time.

Reply:
I know there is DNS caching, not aware that we have JVM or os caching for this, if we tested this is trivial, I totally agree we should keep this simple.

Reply:
IMHO JVM caching is overkill too. It's not application's responsibility to deal with DNS re-resolution, because it shouldn't happen in failure scenarios. On the flipside to avoid DNS server flooding some caching in the app would be reasonable and JVM is a standard way of doing that.